### DISCUSSION TITLE
Hoping to change the text line height/ leading?

### DISCUSSION THREAD

**Discussion by drpeck** [SixLabors/ImageSharp.Drawing#112] [Q&A] [answered]
I'm outputting text like so:
```
var textOptions = new TextGraphicsOptions
{
    TextOptions = new TextOptions
    {
        WrapTextWidth = 1000        
    }
};

image.Mutate(c => c.DrawText(textOptions, myLongText, regularFont, textColour, new Point(0,0)));
```

It's all good, but to match the design I've been given, I need to increase the space between the wrapped lines of text. I can't see any method of doing this. Is there a way?

**Comment by antonfirsov** [comment]
As far as I'm aware, there is none today, and you have to implement your custom layout with `TextMeasurer`.

I think this might be a valid feature request for SixLabors.Fonts, @tocsoft thoughts?

**Reply by drpeck** [reply]
Are there any examples of this? All I'm imagining is a loop that adds words to `TextMeasurer.Measure("Text to measure", options);` until you hit the limit, and then output the text. You'd need output another line further down (etc. etc.).

I assume 'custom layouts' aren't a concept that I've missed.

**Comment by drpeck** [comment]
I have a commercial license if this would make any difference to the prioritization of the feature request.

**Reply by drpeck** [reply]
It would actually be really useful to know, even if this is accepted, what sort of timeframes I'd be looking at. I need to find a solution in a matter of weeks. I'm guessing that it's not reasonable to expect that this could be implemented in this timeframe anyway.

I'd be happy to have a look at this myself, but I haven't the first clue where I'd start.

**Reply by antonfirsov** [reply]
@drpeck that would be great! Personally, I have difficulties with time now, and can't promise anything for December, but I think we can make it as a cooperation.

For now what I can say is that the logic is somewhere in this code:
https://github.com/SixLabors/Fonts/blob/4beaff16420c706a10c293cca6e1c1037a420d36/src/SixLabors.Fonts/TextLayout.cs#L25-L276

You probably need a way to increase this offset with a parameter:
https://github.com/SixLabors/Fonts/blob/4beaff16420c706a10c293cca6e1c1037a420d36/src/SixLabors.Fonts/TextLayout.cs#L279

I suggest to clone SixLabors/Fonts, add a new property to [`RenderOptions`](https://github.com/SixLabors/Fonts/blob/4beaff16420c706a10c293cca6e1c1037a420d36/src/SixLabors.Fonts/RendererOptions.cs) and try if it helps.

**Reply by antonfirsov** [reply]
@drpeck please take a look at  SixLabors/Fonts#153 if you have a chance!

**Reply by drpeck** [reply]
I'm having a really hard time working out how to download all different changes required. However, once the changes are added to `TextOptions` this looks like exactly what I need. 👍👍

**Reply by antonfirsov** [reply]
You can try it as soon as #116 is merged, stay tuned.

**Reply by JimBobSquarePants** [reply]
@drpeck Everything is merged and deployed to our build server. 

If you add the following source to your NuGet.config you can update to `1.0.0-beta11.2` which contains the new property.
```
https://www.myget.org/F/sixlabors/api/v3/index.json
```

**Reply by drpeck** [reply]
Gotcha. 

This works perfectly, from my tests. Thanks for accepting the request but also the speedy acceptance and implementation. 👍👍

**Reply by JimBobSquarePants** [reply]
Perfect! Thanks for your support 👍

**Comment by antonfirsov** [comment, marked as answer]
Opened SixLabors/Fonts#152 to track the feature request. Seems like @JimBobSquarePants can also help here, but the more eyes on the code the faster we finish it.

**Reply by antonfirsov** [reply]
Also opened #114.
